server/src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -384,7 +384,9 @@ public static void writeTotalHits(StreamOutput out, TotalHits totalHits) throws
      * by shard for sorting purposes.
      */
     public static void writeTopDocsIncludingShardIndex(StreamOutput out, TopDocs topDocs) throws IOException {
-        if (topDocs instanceof TopFieldGroups topFieldGroups) {
+        if (topDocs == null) {
```

Comment:
I wonder if we should add BwC for this? I guess no need to if we merge to 8.19 right away since those tests didn't break in the first place and therefore we know that this branch is never taken :)

Reply:
I was assuming that we are good because we are behind a feature flag. There is no roll-out made with the feature enabled, hence no bw comp guarantees?

Reply:
I think you can by the same logic drop the other batched execution version constant once you're done backporting. That's actually fewer PRs I think? You just add the constant here, then backport, then adjust the version usage in main and remove the existing batched exec version constant in one PR.
-> 3 steps vs. 2 turn it off Prs + 1 merge and then two more for flipping the feature on again.

Reply:
will do so , thanks